onadata/libs/permissions.py

```diff
@@ -300,47 +347,57 @@ class OwnerRole(Role):
 # Memoize a class to permissions dict.
 for role in ROLES.values():
     role.class_to_permissions = defaultdict(list)
-    [role.class_to_permissions[k].append(p) for p, k in role.permissions]
+    for p, k in role.permissions:
+        role.class_to_permissions[k].append(p)
 
 
 def is_organization(obj):
-    """Some OrganizationProfiles have a pointer to the UserProfile, but no
+    """
+    Some OrganizationProfiles have a pointer to the UserProfile, but no
     UserProfiles do. Check for that first since it avoids a database hit.
     """
     try:
-        hasattr(obj, 'userprofile_ptr') or obj.organizationprofile
-        return True
+        return (hasattr(obj, 'userprofile_ptr') or
+                obj.organizationprofile is not None)
     except OrganizationProfile.DoesNotExist:
         return False
 
 
 def get_role(permissions, obj):
-    for role in reversed(ROLES_ORDERED):
-        if role.has_role(permissions, obj):
-            return role.name
+    """
+    Return the user role for the given obj permissions.
+    """
+    for _role in reversed(ROLES_ORDERED):
+        if _role.has_role(permissions, obj):
+            return _role.name
+    return None
 
 
 def get_role_in_org(user, organization):
+    """
+    Return the user role in the organization.
+    """
     perms = get_perms(user, organization)
 
     if 'is_org_owner' in perms:
```

Comment:
why are these lines even necessary? shouldn't line 385 alone return the correct role?

Reply:
I suppose it was not possible to return an `owner` role from the line 385. I will check if that has changed since this section is 4 years old code.

Reply:
got it, we can move any work here to this issue too, https://github.com/onaio/onadata/issues/1236, about a general permissions namespace refactor